guides/plugins/plugins/content/cms/real-world-cms/add-real-world-cms-element.md

```diff
@@ -0,0 +1,460 @@
+
+The twig file for your element:
+
+{% code title="<plugin root>/src/Resources/app/administration/src/module/sw-cms/elements/cms-button/component/sw-cms-el-cms-button.html.twig" %}
+{% raw %}
+```markup
+{% block sw_cms_element_ninja_cms_button %}
+	<div class="sw-cms-el-ninja-cms-button" v-model="element.config.buttonAlign.value" :style="buttonAlignStyle">
+		<a href="{{element.config.url.value}}" target="_blank" v-model="element.config.title.value" @input="onInputText">
+			<button class="sw-el-ninja-btn" :style="buttonStyles">{{element.config.title.value}}</button>
+		</a>
+	</div>
+{% endblock %}
+```
+{% endraw %}
+{% endcode %}
+
+And of course you also have a .scss file for your element in the component folder:
+
+{% code title="<plugin root>/src/Resources/app/administration/src/module/sw-cms/elements/cms-button/component/sw-cms-el-cms-button.scss" %}
+```css
+.sw-cms-el-ninja-cms-button {
+    display: flex;
+    width: 100%;
+
+    button {
+        outline: none;
+    }
+
+}
+
+.sw-el-ninja-btn {
+    padding: 1rem;
+}
+
+.sw-el-ninja-btn:focus {
+    outline: none;
+}
+```
+{% endcode %}
+
+
+Let's move on with the preview component.
+
+### Element preview component
+
+The preview directory is not much different, we also have 3 files here. Let's start with the `index.js` again:
+
+{% code title="<plugin root>/src/Resources/app/administration/src/module/sw-cms/elements/cms-button/preview/index.js" %}
+```javascript
+import template from './sw-cms-el-preview-cms-button.html.twig';
+import './sw-cms-el-preview-cms-button.scss';
+
+const { Component } = Shopware;
+
+Component.register('sw-cms-el-preview-cms-button', {
+    template
+});
+```
+{% endcode %}
+
+We registered the preview now we only have to do the template and some styles. So we create a new file `sw-cms-el-preview-cms-button.html.twig`.
+
+{% code title="<plugin root>/src/Resources/app/administration/src/module/sw-cms/elements/cms-button/preview/sw-cms-el-preview-cms-button.html.twig" %}
+{% raw %}
+```markup
+{% block sw_cms_element_ninja_button_preview %}
+	<div class="ninja-button-preview">
+		<button class="ninja-btn">Button</button>
+	</div>
+{% endblock %}
+
+```
+{% endraw %}
+{% endcode %}
+
+We only display a simple button right here. Let's add some styles so people can actually take a look at it:
+
+{% code title="<plugin root>/src/Resources/app/administration/src/module/sw-cms/elements/cms-button/preview/sw-cms-el-preview-cms-button.scss" %}
+```css
+.ninja-button-preview {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    height: 100%;
+
+    .ninja-btn {
+        background-color: #4492ed;
+        border: none;
+        color: #fff;
+        padding: 14px 50px;
+        text-align: center;
+        text-decoration: none;
+        display: inline-block;
+        font-size: 16px;
+        margin: 30px 0;
+    }
+}
+```
+{% endcode %}
+
```

Comment:
Now I created a preview element but I don't know where I will see this. Is this part of the required knowledge?

Reply:
If you followed the real-world cms block guide you should see a text element if you drag & drop it in the shopping experiences. So maybe I should add a little sentence like here as well where you can see that element?

But I guess it's clear if you followed the first guide. What are your thoughts?

Reply:
There are in sum like 5 elements I am building. And when no one is telling me why we need a preview I could think it is the preview in the editor. But it is just the preview in the selection screen. Might be confusing. I just don't like the naming shopware used here in the beginning to I think it should always be mentioned where you can see your recent changes so you can follow along.

Reply:
Let me add a short explanation here as well. In first place I just wanted to add images so it's 100% clear but I think images are not possible right?